janusgraph-core/src/main/java/org/janusgraph/core/attribute/Geoshape.java

```diff
@@ -16,22 +16,38 @@
 
 import com.google.common.base.Preconditions;
 import com.google.common.primitives.Doubles;
-import com.spatial4j.core.context.SpatialContext;
+import com.spatial4j.core.context.SpatialContextFactory;
+import com.spatial4j.core.context.jts.DatelineRule;
+import com.spatial4j.core.context.jts.JtsSpatialContext;
+import com.spatial4j.core.context.jts.JtsSpatialContextFactory;
 import com.spatial4j.core.distance.DistanceUtils;
+import com.spatial4j.core.io.jts.JtsBinaryCodec;
+import com.spatial4j.core.io.jts.JtsGeoJSONReader;
+import com.spatial4j.core.io.jts.JtsGeoJSONWriter;
+import com.spatial4j.core.io.jts.JtsWKTReader;
+import com.spatial4j.core.io.jts.JtsWKTWriter;
+import com.spatial4j.core.shape.Circle;
+import com.spatial4j.core.shape.Rectangle;
 import com.spatial4j.core.shape.Shape;
 import com.spatial4j.core.shape.SpatialRelation;
 import org.janusgraph.diskstorage.ScanBuffer;
 import org.janusgraph.diskstorage.WriteBuffer;
 import org.janusgraph.graphdb.database.idhandling.VariableLong;
-import org.janusgraph.graphdb.relations.RelationIdentifier;
-import org.apache.commons.lang.builder.HashCodeBuilder;
+
+import com.spatial4j.core.shape.jts.JtsGeometry;
+import com.vividsolutions.jts.geom.Coordinate;
+import com.vividsolutions.jts.geom.Geometry;
+import com.vividsolutions.jts.geom.GeometryFactory;
```

Comment:
There are a few specialty functions in there that use JTS. I'll need to verify/test but if it's made an optional dependency it would still compile and be used in tests but wouldn't be included in distributions (at least not through janusgraph-core). Is this acceptable? In this way users of relevant (e.g. polygon) features could manually add JTS to their JanusGraph classpath but other users would be unaffected.